pkg/odo/genericclioptions/runnable.go

```diff
@@ -3,21 +3,39 @@ package genericclioptions
 import (
 	"flag"
 	"fmt"
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/segment"
+	"k8s.io/klog"
 	"os"
+	"time"
 
 	"github.com/openshift/odo/pkg/log"
 	"github.com/openshift/odo/pkg/odo/util"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 )
 
+var (
+	segmentClient *segment.Client
+	//ctx context.Context
```

Comment:
commented code?

Reply:
Yes. I kept it for the future. I am not sure if we want to use context as they have done in CRC. In case we do, this would act as a starting point. I will remove it later in case we decide context is not required.

CRC is using context to pass some values to segment. For e.g. w.r.t to odo, if we decide we want to pass values such as `odo preference set buildtimeout 5`, values being `buildtimeout` and `5`, we might need to use context.

Reply:
We would need context to cancel a segment upload when someone `Sigkills` odo? just thinking out loud, if segment uploads are not async in nature then we would need to have a functionality to breakout of a stuck segment upload?

Reply:
I checked the source code for segment's go client and it is asynchronous as far as I can tell, but I will do some more research and make sure it is.

Reply:
It is async.